Source/Utility/DirectoryValueTree.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <JuceHeader.h>
+#include "DirectoryDataProperties.h"
+#include "../Utility/LambdaThread.h"
+#include "../Utility/ValueTreeMonitor.h"
+#include "../Utility/WatchDogTimer.h"
+
+class DirectoryValueTree : public juce::Thread,
+                           private juce::Timer,
+                           private juce::AsyncUpdater
+{
+public:
+    DirectoryValueTree ();
+    ~DirectoryValueTree ();
+
+    void init (juce::ValueTree rootPropertiesVT);
+    juce::ValueTree getDirectoryDataPropertiesVT ();
+
+private:
+    enum class ScanType
+    {
+        checkForUpdate,
+        fullScan
+    };
+    enum class TaskManagementState
+    {
+        idle,
+        startScan,
+        scanning,
+        startCheck,
+        checking,
+    };
+    WatchdogTimer timer; // TODO - remove when not needed, ie. when done measuring things
```

Comment:
I don't think this is actually used?

Reply:
I think I will put it in a compile time macro, along with the corresponding code this uses it in DirectoryValueTree::scanDirectory, so that I can re-enable timing measurements at a later date. I'm kinda cheating using a WatchdogTimer to measure code, lol. But I had it available.